clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1056,8 +1056,20 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
         DepFile = getDependencyFileName(Args, Inputs);
         C.addFailureResultFile(DepFile, &JA);
       }
-      CmdArgs.push_back("-dependency-file");
-      CmdArgs.push_back(DepFile);
+      // for host compile, if OPT_MMMD is on, default to create *.d.host file
+      // and merge it with *.d (kernel dep) file in DependencyFile.cpp
+      Arg *ArgMMD = Args.getLastArg(options::OPT_MMMD);
+      auto at = getToolChain().getAuxTriple();
+      if (ArgMMD && !at && std::string(DepFile) != "-") {
+        SmallString<128> NewDepFile(DepFile);
+        NewDepFile.append(".host");
```

Comment:
This name may conflict with other files. Is there a way to avoid the conflict? How about add CUID as part of postfix?

Reply:
this patten of ".host" is an interface for DependencyFileGenerator::outputDependencyFile
so if we chang it to some CUID, we now do not get any good way to let host compiler know if we need to merge two or more dep file.
I will think over it, and try to find some way to fixed it.